pip/src/interpreter.rs

```diff
@@ -207,6 +207,17 @@ impl Interpreter {
         }
     }
 
+    fn rerun(&mut self, py: Python, callback: Option<PyObject>) -> PyResult<PyObject> {
```

Comment:
It feels a little ugly to keep duplicating code for the == 1 or > 1 case. Might it be possible to just make `entry_expr` on `run` optional, and if you don't pass it we just try to re-run the last graph (i.e. call `run_last_expr` if not provided).

Reply:
I debated about that and was on the fence. I don't know why I lean away from having lots of optional parameters to a function, but as you point out this case could really benefit from simplifying. I'll make the switch!